procedures/unit-testing-basics.ipf

```diff
@@ -669,21 +691,33 @@ End
 
 /// Prints an informative message that the test failed
 /// @param prefix string to be added at the beginning
-Function PrintFailInfo([prefix])
-	String prefix
+Function PrintFailInfo(expectedFailure)
+	variable expectedFailure
 
-	string str
+	string prefix, str
+	variable size, index
+	WAVE/T wv = GetFailedProcWave()
 
 	DFREF dfr = GetPackageFolder()
 	SVAR/SDFR=dfr message
 	SVAR/SDFR=dfr status
 	SVAR/SDFR=dfr type
 
-	prefix = SelectString(ParamIsDefault(prefix), prefix, "")
+	prefix = SelectString(expectedFailure, "", "Expected Failure: ")
 
 	str = getInfo(0)
 	message = prefix + status + " " + str
 
+	if(!expectedFailure)
+		index = NumberByKey(TC_SUMMARY_LENGTH_KEY, note(wv))
+		size = DimSize(wv, 0)
+		if(index >= size)
+			Redimension/N=(size * 2) wv
```

Comment:
Just a comment from the sidelines. Why not use EnsureLargeEnoughWaveSimple?

Reply:
> then this method redimensions 5 times

That's not right. The starting size is already 16 (set by a constant above). So it will only redimension once.

> using the linear increase you have to redimension only once.

That depends on the start size a lot. Lets say both (linear and exponential) use the same start size. If you have less than triple of entries of start size you have the same number of resizes. But if the list grow more you will have more resizes in a linear increase than in an exponential increase.

![grafik](https://user-images.githubusercontent.com/12125053/201063670-74b53154-0ff7-4b3f-939d-a58a7653e205.png)
*(In this diagram the start size is one chunk (an arbitrary number of entries).)*

And that's the reason why many list or vector implementation use the exponential increase (or decrease) with a reasonable start size.

> Why not use EnsureLargeEnoughWaveSimple?

I didn't know there was something like that but I will use it. The problem here is that is uses the linear approach.

Reply:
If Michael agrees I'm okay with teaching EnsureLargeEnoughWaveSimple to select linear/exponential. One important point here is that we want to use a generic function for things like that. Solve it once properly and then reuse it!

Reply:
Yes, lets teach EnsureLargeEnoughWaveSimple the exponential increase and use that function then.

Reply:
I've added a new commit to do this.